cmd/admin-revoker/main.go

```diff
@@ -211,7 +238,135 @@ func (r *revoker) revokeByReg(ctx context.Context, regID int64, reasonCode revoc
+}
+
+// blockByPrivateKey blocks future issuance for certificates with a a public key
+// matching the SubjectPublicKeyInfo hash generated from the PublicKey embedded
+// in privateKey. The embedded PublicKey will be verified as an actual match for
+// the provided private key before any blocking takes place. This method does
+// not revoke any certificates directly. However, 'bad-key-revoker', which
+// references the 'blockedKeys' table, will eventually revoke certificates with
+// a matching SPKI hash.
+func (r *revoker) blockByPrivateKey(ctx context.Context, privateKey string) error {
+	_, publicKey, err := privatekey.Load(privateKey)
+	if err != nil {
+		return err
+	}
+
+	spkiHash, err := getPublicKeySPKIHash(publicKey)
+	if err != nil {
+		return err
+	}
+
+	u, err := user.Current()
+	if err != nil {
+		return err
+	}
+
+	req := &sapb.AddBlockedKeyRequest{
+		KeyHash:   spkiHash,
+		Added:     r.clk.Now().UnixNano(),
+		Source:    "admin-revoker",
+		Comment:   fmt.Sprintf("blocked by %s", u),
+		RevokedBy: 0,
+	}
+
+	_, err = r.sac.AddBlockedKey(ctx, req)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// revokeByPrivateKey revokes all certificates with a public key matching the
+// SubjectPublicKeyInfo hash generated from the PublicKey embedded in
+// privateKey. The embedded PublicKey will be verified as an actual match for the
+// provided private key before any revocation takes place. The provided key will
+// not be added to the 'blockedKeys' table. This is done to avoid a race between
+// 'admin-revoker' and 'bad-key-revoker'. You MUST call blockByPrivateKey after
+// calling this function, on pain of violating the BRs.
+func (r *revoker) revokeByPrivateKey(ctx context.Context, privateKey string) error {
+	_, publicKey, err := privatekey.Load(privateKey)
+	if err != nil {
+		return err
+	}
+
+	spkiHash, err := getPublicKeySPKIHash(publicKey)
+	if err != nil {
+		return err
+	}
+
+	matches, err := r.getCertsMatchingSPKIHash(spkiHash)
+	if err != nil {
+		return err
+	}
+
+	for i, match := range matches {
+		resp, err := r.sac.GetCertificateStatus(ctx, &sapb.Serial{Serial: match})
+		if err != nil {
+			return fmt.Errorf(
+				"failed to get status for serial %q. Entry %d of %d affected certificates: %w",
+				match,
+				(i + 1),
+				len(matches),
+				err,
+			)
+		}
+
+		if resp.Status != string(core.OCSPStatusGood) {
+			r.log.AuditInfof("serial %q is already revoked, skipping", match)
+			continue
+		}
+
+		err = r.revokeBySerial(ctx, match, revocation.Reason(1), true)
+		if err != nil {
+			return fmt.Errorf(
+				"failed to revoke serial %q. Entry %d of %d affected certificates: %w",
+				match,
+				(i + 1),
+				len(matches),
+				err,
+			)
+		}
+	}
+	return nil
+}
+
+func (r *revoker) spkiHashInBlockedKeys(spkiHash []byte) (bool, error) {
+	var count int
+	err := r.dbMap.SelectOne(&count, "SELECT COUNT(*) as count FROM blockedKeys WHERE keyHash = ?;", spkiHash)
```

Comment:
```suggestion
	err := r.dbMap.SelectOne(&count, "SELECT COUNT(*) as count FROM blockedKeys WHERE keyHash = ?", spkiHash)
```

We don't need to put `;` in our queries. The DB driver handles it. Same below.